aiida/manage/database/delete/nodes.py

```diff
@@ -37,70 +41,82 @@ def delete_nodes(pks, verbosity=0, dry_run=False, force=False, **kwargs):
 
     :param pks: a list of the PKs of the nodes to delete
     :param bool force: do not ask for confirmation to delete nodes.
-    :param int verbosity: 0 prints nothing,
+    :param int verbosity: (deprecated) 0 prints nothing,
                           1 prints just sums and total,
                           2 prints individual nodes.
-
     :param kwargs: graph traversal rules. See :const:`aiida.common.links.GraphTraversalRules` what rule names
         are toggleable and what the defaults are.
     :param bool dry_run:
         Just perform a dry run and do not delete anything. Print statistics according
         to the verbosity level set.
     :param bool force:
         Do not ask for confirmation to delete nodes.
+
+    .. deprecated:: 1.4.0
+            Argument `verbosity` will be removed in `v2.0.0`. Control verbosity by setting the log level of the "verdi"
+            logger instead.
```

Comment:
I am not sure I agree with this. It is perfectly reasonable to call this from a shell or any other way that is not necessarily `verdi`. And in those cases you also may want to control the verbosity. I agree that the current implementation is not ideal, but instead of completely removing it, shouldn't we provide another way of setting it? Also, the way it is currently tied directly to the command line interface logger is not the way to go.

Reply:
Well, one could argue that the function is already tied to the cli because it is printing stuff to stdout - the most flexible option would probably be to pass it a logger as an argument and then use that to log any messages. 

The cli would pass the verdi logger, which could also be the default, but other callers may choose to pass a different logger.
How does that sound?

Reply:
Note: the current implementation of the `export` function asks users to just set the log level of the `EXPORT_LOGGER` if desired.

https://github.com/aiidateam/aiida-core/blob/d29fb3be78c9b7be2e495b287c6dca8960bfe83d/aiida/tools/importexport/dbexport/__init__.py#L95-L101

That should also work here.